[Gao Haochun] iP

Harry-Gao-H opened the pull request:
## Duke 10 Plus

> We're living the future so
> the present is our past.

Duke 10 Plus frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* SUPER FAST to use

All you need to do is,
1. download it from [here](www.google.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Harry-Gao-H committed:
t:

Harry-Gao-H committed:
Add list function

Harry-Gao-H committed:
add Task class and mark as done functionality

Harry-Gao-H committed:
Add support for tracking three types of tasks

Harry-Gao-H committed:
Semi-automate the testing of Duke

Harry-Gao-H committed:
Handle some invalid format

Harry-Gao-H committed:
Add javadoc

Harry-Gao-H committed:
Add delete functionality

Harry-Gao-H committed:
Formalize the coding conventions.

Harry-Gao-H committed:
Implement record export

Harry-Gao-H committed:
Implement Save functionality

Harry-Gao-H committed:
Implement Dates functionality

Harry-Gao-H committed:
Merge branch 'branch-Level-7'

Harry-Gao-H committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Task.java

Harry-Gao-H committed:
(Uncompleted) Improve OOP

Harry-Gao-H committed:
Make the code more OOP

Harry-Gao-H committed:
Divide classes into packages

Harry-Gao-H committed:
Add JUnit tests to test the behavior of the code.

Harry-Gao-H committed:
Fix the bug that cannot exit.

Harry-Gao-H committed:
Enable "Find a task by searching for a keyword"

Harry-Gao-H committed:
Add JavaDoc comments to the code

Harry-Gao-H committed:
Tweak the code to comply with a coding standard

Harry-Gao-H committed:
Merge branch 'branch-Level-9'

Harry-Gao-H committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java

Harry-Gao-H committed:
Merge branch 'branch-A-CodingStandard'

LMAOboxhack commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,40 @@
+        this.isDone = true;
+    }
+
+    String encoding() {
```
```suggestion
    String encode() {
```
Would it be better to use the verb version of the word instead, as outlined in the [Code Quality: Naming](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/basic/nounsAndVerbsAsNames/) section of the textbook?

LMAOboxhack commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,222 @@
+     * @return a string showing the added task and number of tasks
+     */
+     static String addList(String input, TaskList taskList) {
+         ArrayList<Task> lst = taskList.getTasks();
+         ArrayList<String> validType = new ArrayList<>(
```
```suggestion
         ArrayList<Task> tasks = taskList.getTasks();
         ArrayList<String> validTypes = new ArrayList<>(
```
Would it be better to name your variables in plural form instead, to show that they are collections of items? See [this](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/basic/nounsAndVerbsAsNames/) section of the textbook.

LMAOboxhack commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+import java.time.format.DateTimeParseException;
+
+class Deadline extends Task {
+    protected String by;
```
Would it be good to use a more descriptive variable name here? Right now "by" is not very indicative of the variable's function in the class. See '[Use name to explain](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/)' from the textbook.

LMAOboxhack commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,222 @@
+     * @return the messages
+     */
+    static String doTask(String index, TaskList taskList) {
+        int idx;
```
```suggestion
        int indexToMarkComplete;
```
Would it be better to use standard words in your variable naming here? Also might be good to specify a more descriptive variable name. See '[Use standard words](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/basic/useStandardWords/)' & '[Use name to explain](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/)' in the textbook.

LMAOboxhack commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public ArrayList<Task> loadTasks() throws DukeException {
+        File f = new File(filePath);
```
```suggestion
        File fileToLoad = new File(filePath);
```

Would it be better to specify a more descriptive variable name here? See '[Use standard words](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/basic/useStandardWords/)' & '[Use name to explain](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/)' in the textbook.

LMAOboxhack reviewed (commented):
Looks good 👍 just some minor issues such as method and variable naming.

Harry-Gao-H committed:
Merge branch 'add-gradle-support'

Harry-Gao-H committed:
Install gradle

Harry-Gao-H committed:
Improve code quality

Harry-Gao-H committed:
Add JavaFX basics

Harry-Gao-H committed:
Add a GUI to Duke

Harry-Gao-H committed:
Add javadoc

mukundrs reviewed (commented):
Your code looks really good to me! Just a few nitpicks here and there but nothing too imp. Some classes and functions are missing javadocs. While they do seem redundant sometimes, its generally good practice to include them anyways. There were a few lines of commented out code which personally I would remove so that it improves readability. Apart from those nitpicks  I thought your code was very easy to read and very well done! 😄

mukundrs commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+    private Label dialog;
+    @FXML
+    private ImageView displayPicture;
+
```
Missing javadocs for constructor

mukundrs commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+        getChildren().setAll(tmp);
+        setAlignment(Pos.TOP_LEFT);
+    }
+
```
Missing javadocs for function

mukundrs commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
```
Missing javadocs for functions

mukundrs commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,197 @@
+import duke.TaskList;
+import duke.Ui;
+import javafx.fxml.FXML;
+
```
Missing javadocs for class description

mukundrs commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,197 @@
+            taskList = new TaskList();
+        }
+    }
+
```
Missing javadocs for function

mukundrs commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,197 @@
+//
+//        //Part 3. Add functionality to handle user input.
+//        sendButton.setOnMouseClicked((event) -> {
+//            handleUserInput();
+//        });
+//
+//        userInput.setOnAction((event) -> {
+//            handleUserInput();
+//        });
+//    }
+//
+//    /**
+//     * Iteration 1:
+//     * Creates a label with the specified text and adds it to the dialog container.
+//     * @param text String containing text to add
+//     * @return a label with the specified text that has word wrap enabled.
+//     */
+//    @FXML
+//    private Label getDialogLabel(String text) {
+//        // You will need to import `javafx.scene.control.Label`.
+//        Label textToAdd = new Label(text);
+//        textToAdd.setWrapText(true);
+//
+//        return textToAdd;
+//    }
+//
+//    /**
+//     * Iteration 2:
+//     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+//     * the dialog container. Clears the user input after processing.
+//     */
+//    private void handleUserInput() {
+//        Label userText = new Label(userInput.getText());
+//        Label dukeText = new Label(getResponse(userInput.getText()));
+//        dialogContainer.getChildren().addAll(
+//                DialogBox.getUserDialog(userText, new ImageView(user)),
+//                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+//        );
+//        userInput.clear();
+//    }
```
Would personally remove these comments to improve readability

mukundrs commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,197 @@
+//        );
+//        userInput.clear();
+//    }
+    Ui getUi() {
```
Missing javadocs for function

mukundrs commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,29 @@
+public class Main extends Application {
+
+    private Duke duke = new Duke();
+
```
Missing javadocs for function

mukundrs commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,56 @@
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
```
Missing javadocs

mukundrs commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,56 @@
+                DialogBox.getDukeDialog(output, dukeImage)
+        );
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+class Deadline extends Task {
+    protected String by;
+    protected LocalDate date;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+            date = null;
+        }
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+            date = null;
+        }
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+    LocalDate getDate() {
+        return date;
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+    String encoding() {
+        return "D&&" + super.encoding() + "&&" + by;
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,51 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,51 @@
+class Event extends Task {
+    protected String at;
+    protected LocalDate date;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,51 @@
+            date = null;
+        }
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,51 @@
+            date = null;
+        }
+    }
+
+    @Override
+    LocalDate getDate() {
+        return date;
+    }
+
+    @Override
+    String encoding() {
+        return "E&&" + super.encoding() + "&&" + at;
+    }
+
+    @Override
+    public String toString() {
+        String time;
+        if (date == null) {
+            time = at;
+        } else {
+            time = date.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        }
+        return "[E]" + super.toString() + " (at: " + time + ")";
+    }
+}
```
Missing javadocs

mukundrs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,221 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,221 @@
+            return null;
+        }
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,221 @@
+
+        return s.toString();
+    }
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,58 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,40 @@
+package duke;
+
+import java.time.LocalDate;
+
+class Task {
+    protected String content;
+    protected boolean isDone;
+    Task(String content) {
+        this.content = content;
+        this.isDone = false;
+    }
+
+    Task(String content, boolean isDone) {
+        this.content = content;
+        this.isDone = isDone;
+    }
+
+    void setDone() {
+        this.isDone = true;
+    }
+
+    String encoding() {
+        return (isDone ? "Done" : "InProgress") + "&&" + content;
+    }
+
+    String getContent() {
+        return content;
+    }
+
+    LocalDate getDate() {
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("[%s] %s", isDone ? "X" : " ", this.content);
+    }
+
+
+}
```
Missing javadocs

mukundrs commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+package duke;
+
+import java.util.ArrayList;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    public TaskList() {
```
Missing javadocs and would personally have an empty line in between tasks and tasklist()

mukundrs commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+    public TaskList() {
+        tasks = new ArrayList<Task>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    Task getTask(int index) {
+        assert index < tasks.size();
+        return tasks.get(index);
+    }
+
+    ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    void deleteTask(int index) throws DukeException {
+        try {
+            tasks.remove(index);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("The task does not exit");
+        }
+    }
+
+    void addTask(Task t) {
+        tasks.add(t);
+    }
+
+}
```
Missing javadocs

mukundrs commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,21 @@
+package duke;
+
+class Todo extends Task {
+    Todo(String content) {
+        super(content);
+    }
+
+    Todo(String content, boolean isDone) {
+        super(content, isDone);
+    }
+
+    @Override
+    String encoding() {
+        return "T&&" + super.encoding();
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
```
Missing javadocs

mukundrs commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,70 @@
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
```
Missing javadocs

mukundrs commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,70 @@
+                + "    ____________________________________________________________\n";
+        return output;
+    }
+
```
Missing javadocs

mukundrs reviewed (commented):
Sorry i messed up with the comments on my last review lololol. Here are my comments!

Harry-Gao-H committed:
Merge branch 'branch-Level-10'

Harry-Gao-H committed:
Use assert feature to document important assumptions that should hold at various points in the code.

Harry-Gao-H committed:
Merge pull request #1 from Harry-Gao-H/branch-Assertions

Use assert feature to document important assumptions that should hold…

Harry-Gao-H committed:
Restructure the packages

Harry-Gao-H committed:
Improve code quality (imcompleted)

Harry-Gao-H committed:
Some functionality of Parser is moved to Command to improve Code Quality.

The methods in Parser was very long.

Command interface and a series of classes implementing the interface is added.
The parser now passes command to Duke.

Shorter methods with clear naming is more understandable and easier to maintain.

Harry-Gao-H committed:
Merge pull request #2 from Harry-Gao-H/branch-CodeQuality

Branch code quality

Harry-Gao-H committed:
Add help extension

Harry-Gao-H committed:
Improve code quality

Harry-Gao-H committed:
Use the Streams feature

Harry-Gao-H committed:
Improve the GUI

Harry-Gao-H committed:
Improve the error message

Harry-Gao-H committed:
Add a User Guide to the project

Harry-Gao-H committed:
Modify user guide and create jar file

Harry-Gao-H committed:
Give credit for reused work

Harry-Gao-H committed:
Update readme

Harry-Gao-H committed:
Modify output format
